apollo-federation/src/operation/mod.rs

```diff
@@ -2325,12 +2327,14 @@ impl SelectionSet {
             return Ok(Conditions::Boolean(false));
         };
         let conditions = first_selection.conditions()?;
-        for selection in selections {
-            if selection.conditions()? != conditions {
-                return Ok(Conditions::Boolean(true));
-            }
+        if selections
+            .map(|selection| selection.conditions())
+            .ok_and_any(|cond| cond != conditions)?
+        {
+            Ok(Conditions::Boolean(true))
+        } else {
+            Ok(conditions)
```

Comment:
I’m not super convinced this is an improvement over the `for` loop but 🤷

Reply:
Ya, I was probably a little overly zealous here.